File: code-coaches/easy/land-ho/land_ho.rb
def get_waiting_time(num_people_ahead)
  trip_duration = 10
  boat_capacity = 20

  # add 1 to account for yourself
  num_boats_needed = ((num_people_ahead + 1) / boat_capacity.to_f).ceil

  # trip_duration * 2 because of round trip
  # num_boats_needed - 1 to not include our boat
  waiting_time = (num_boats_needed - 1) * trip_duration * 2

  total_waiting_time = waiting_time + trip_duration
end

def main()
  num_people_ahead = gets.chomp.to_i
  puts get_waiting_time(num_people_ahead)
end

if __FILE__ == $0
  main()
end

File: code-coaches/easy/land-ho/LandHo.cs
using System;

public class LandHo
{
    static int GetWaitingTime(int numPeopleAhead)
    {
        int tripDuration = 10;
        int boatCapacity = 20;

        // add 1 to account for yourself
        int numBoatsNeeded = (int) Math.Ceiling((double) (numPeopleAhead + 1) / boatCapacity);

        // tripDuration * 2 because of round trip
        // numBoatsNeeded - 1 to not include our boat
        int waitingTime = (numBoatsNeeded - 1) * tripDuration * 2;

        int totalWaitingTime = waitingTime + tripDuration;
        return totalWaitingTime;
    }

    static void Main(string[] args)
    {
        int numPeopleAhead = int.Parse(Console.ReadLine());
        Console.WriteLine(GetWaitingTime(numPeopleAhead));
    }
}
